### PR TITLE
pytorch tensor truth value

### PR DESCRIPTION
**Pull request by noskill**
Wrapper truth value class for python object.  
TTruthValue c++ class wraps pytorch tensor object to query for mean and confidence, so that it can be accessed from both c++ and python.

see issue https://github.com/singnet/opencog/issues/70

### CODE REVIEW

File: opencog/cython/opencog/atomspace.pxd [hunk 1]
```diff
@@ -103,6 +103,23 @@ cdef class TruthValue(Value):
     cdef tv_ptr* _tvptr(self)
 
 
+cdef extern from  "opencog/atoms/truthvalue/TTruthValue.h" namespace "opencog":
+    ctypedef shared_ptr[const cTTruthValue] ttv_ptr "opencog::TTruthValuePtr"
+    cdef cppclass cTTruthValue "opencog::TTruthValue"(cTruthValue):
+        cTTruthValue(object)
+        strength_t get_mean()
```
Comment (vsbogd) [inline opencog/cython/opencog/atomspace.pxd]:
```suggestion
        strength_t get_mean() except +
```

File: opencog/cython/opencog/atomspace.pxd [hunk 2]
```diff
@@ -103,6 +103,23 @@ cdef class TruthValue(Value):
...
+        confidence_t get_confidence()
```
Comment (vsbogd) [inline opencog/cython/opencog/atomspace.pxd]:
```suggestion
        confidence_t get_confidence() except +
```

File: opencog/atoms/atom_types/atom_types.script [hunk 3]
```diff
@@ -32,6 +32,7 @@ INDEFINITE_TRUTH_VALUE     <- TRUTH_VALUE
 FUZZY_TRUTH_VALUE          <- TRUTH_VALUE
 PROBABILISTIC_TRUTH_VALUE  <- TRUTH_VALUE
 EVIDENCE_COUNT_TRUTH_VALUE <- TRUTH_VALUE
+TTRUTH_VALUE               <- TRUTH_VALUE "TTruthValue"
```
Comment (vsbogd) [inline opencog/atoms/atom_types/atom_types.script]: I suggest name it `TORCH_TRUTH_VALUE ` as `TTRUTH_VALUE` is unclear and ambiguous - usually `T` at the beginning sounds as `Type`.  And rename `TTruthValue` to `TorchTruthValue` as well for clarity.

File: tests/cython/atomspace/test_tensortv.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from opencog.utilities import initialize_opencog, finalize_opencog
+from opencog.type_constructors import *
+
+
+torch_found = True
+try:
+    from opencog.atomspace import TensorTruthValue
+    from torch import tensor
+    import torch
+    TTruthValue = TensorTruthValue
```
Comment (vsbogd) [inline tests/cython/atomspace/test_tensortv.py]: Why did you not renamed classes in test? As test can be used as example of code it can be misleading.

Comment (noskill) [reply tests/cython/atomspace/test_tensortv.py]: I don't see misleading class names in this file. Which one looks misleading to you?

**Pull request merged**